### ISSUE TITLE
[Bug Report][Crash][macOS/iOS] App crash when exit video page and reopen

### ISSUE THREAD

**Issue by DaVinci42** [media-kit/media-kit#498]
### Steps to reproduce:
1. with the following code 
2. home page, click "Open Media"
3. media page, click "Pop"
4. repeat like 20 times, the app will crash

### Dev note:
100% reproduced on macOS and iOS

### Version:
```yaml
media_kit: ^1.1.7
media_kit_video: ^1.1.8
media_kit_libs_video: ^1.0.1
```

### Minimal code to reproduce:
```dart
void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const Scaffold(
        body: SafeArea(child: PageA()),
      ),
    );
  }
}

class PageA extends StatefulWidget {
  const PageA({super.key});

  @override
  State<PageA> createState() => _PageAState();
}

class _PageAState extends State<PageA> {
  @override
  Widget build(BuildContext context) {
    return Center(
      child: ElevatedButton(
        onPressed: () {
          Navigator.of(context).push(
            MaterialPageRoute(builder: (_) => const MediaPage()),
          );
        },
        child: const Text("Open Media"),
      ),
    );
  }
}

class MediaPage extends StatefulWidget {
  const MediaPage({super.key});

  @override
  State<MediaPage> createState() => _MediaPageState();
}

class _MediaPageState extends State<MediaPage> {
  late final player = Player();
  late final controller = VideoController(player);

  @override
  void initState() {
    super.initState();
    MediaKit.ensureInitialized();
    player.open(
        Media(
            'https://user-images.githubusercontent.com/28951144/229373695-22f88f13-d18f-4288-9bf1-c3e078d83722.mp4'),
        play: false);
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Center(
            child: Column(
          mainAxisSize: MainAxisSize.min,
          crossAxisAlignment: CrossAxisAlignment.stretch,
          children: [
            SizedBox(
              width: 200,
              height: 200,
              child: Video(controller: controller),
            ),
            ElevatedButton(
              onPressed: () {
                Navigator.of(context).pop();
              },
              child: const Text("POP"),
            ),
          ],
        )),
      ),
    );
  }
}
```

### Crash log:
```
Thread 35 Crashed:
0   libsystem_kernel.dylib        	       0x190666868 __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x19069dcec pthread_kill + 288
2   libsystem_c.dylib             	       0x1905d62c8 abort + 180
3   libc++abi.dylib               	       0x190656b18 abort_message + 132
4   libc++abi.dylib               	       0x1906468f0 demangling_terminate_handler() + 52
5   libc++abi.dylib               	       0x190655eb4 std::__terminate(void (*)()) + 20
6   libc++abi.dylib               	       0x190655e3c std::terminate() + 36
7   media_kit_native_event_loop   	       0x103e2036c std::__1::default_delete<std::__1::thread>::operator()[abi:v15006](std::__1::thread*) const + 52 (unique_ptr.h:48)
8   media_kit_native_event_loop   	       0x103e20328 std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> >::reset[abi:v15006](std::__1::thread*) + 104 (unique_ptr.h:305)
9   media_kit_native_event_loop   	       0x103e202b0 std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> >::~unique_ptr[abi:v15006]() + 32 (unique_ptr.h:259)
10  media_kit_native_event_loop   	       0x103e160c0 std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> >::~unique_ptr[abi:v15006]() + 28 (unique_ptr.h:259)
11  media_kit_native_event_loop   	       0x103e21d8c std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >::~pair() + 32 (pair.h:40)
12  media_kit_native_event_loop   	       0x103e21d5c std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >::~pair() + 28 (pair.h:40)
13  media_kit_native_event_loop   	       0x103e21d00 void std::__1::allocator_traits<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > >::destroy[abi:v15006]<std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void, void>(std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> >&, std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >*) + 28 (allocator_traits.h:319)
14  media_kit_native_event_loop   	       0x103e21ca0 std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > >::operator()[abi:v15006](std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>*) + 84 (__hash_table:815)
15  media_kit_native_event_loop   	       0x103e21c3c std::__1::unique_ptr<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>, std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > > >::reset[abi:v15006](std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>*) + 104 (unique_ptr.h:305)
16  media_kit_native_event_loop   	       0x103e21bc4 std::__1::unique_ptr<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>, std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > > >::~unique_ptr[abi:v15006]() + 32 (unique_ptr.h:259)
17  media_kit_native_event_loop   	       0x103e20cec std::__1::unique_ptr<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>, std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > > >::~unique_ptr[abi:v15006]() + 28 (unique_ptr.h:259)
18  media_kit_native_event_loop   	       0x103e25340 std::__1::__hash_table<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::__unordered_map_hasher<mpv_handle*, std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::hash<mpv_handle*>, std::__1::equal_to<mpv_handle*>, true>, std::__1::__unordered_map_equal<mpv_handle*, std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::equal_to<mpv_handle*>, std::__1::hash<mpv_handle*>, true>, std::__1::allocator<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > > > >::erase(std::__1::__hash_const_iterator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>*>) + 96 (__hash_table:2403)
19  media_kit_native_event_loop   	       0x103e250ac unsigned long std::__1::__hash_table<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::__unordered_map_hasher<mpv_handle*, std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::hash<mpv_handle*>, std::__1::equal_to<mpv_handle*>, true>, std::__1::__unordered_map_equal<mpv_handle*, std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::equal_to<mpv_handle*>, std::__1::hash<mpv_handle*>, true>, std::__1::allocator<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > > > >::__erase_unique<mpv_handle*>(mpv_handle* const&) + 120 (__hash_table:2435)
20  media_kit_native_event_loop   	       0x103e245d4 std::__1::unordered_map<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> >, std::__1::hash<mpv_handle*>, std::__1::equal_to<mpv_handle*>, std::__1::allocator<std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > > > >::erase[abi:v15006](mpv_handle* const&) + 32 (unordered_map:1360)
21  media_kit_native_event_loop   	       0x103e24430 MediaKitEventLoopHandler::Dispose(long long)::$_1::operator()() const + 116 (media_kit_native_event_loop.cc:137)
22  media_kit_native_event_loop   	       0x103e24388 decltype(std::declval<MediaKitEventLoopHandler::Dispose(long long)::$_1>()()) std::__1::__invoke[abi:v15006]<MediaKitEventLoopHandler::Dispose(long long)::$_1>(MediaKitEventLoopHandler::Dispose(long long)::$_1&&) + 24 (invoke.h:394)
23  media_kit_native_event_loop   	       0x103e24364 void std::__1::__thread_execute[abi:v15006]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, MediaKitEventLoopHandler::Dispose(long long)::$_1>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, MediaKitEventLoopHandler::Dispose(long long)::$_1>&, std::__1::__tuple_indices<>) + 28 (thread:290)
24  media_kit_native_event_loop   	       0x103e24074 void* std::__1::__thread_proxy[abi:v15006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, MediaKitEventLoopHandler::Dispose(long long)::$_1> >(void*) + 84 (thread:301)
25  libsystem_pthread.dylib       	       0x19069e06c _pthread_start + 148
26  libsystem_pthread.dylib       	       0x190698e2c thread_start + 8
```

**Comment by birros** [comment]
```diff
   class _MediaPageState extends State<MediaPage> {
         play: false);
   }
 
+  @override
+  dispose() {
+    player.dispose();
+    super.dispose();
+  }
+
   @override
   Widget build(BuildContext context) {
     return Scaffold(
```

**Comment by DaVinci42** [comment]
@birros 

I did miss `dispose()` in the example above, but actually the issue still exists after adding the func.

**Comment by birros** [comment]
Try with main branch:

```yaml
dependencies:
  media_kit:
    git:
      url: https://github.com/media-kit/media-kit.git
      ref: main
      path: ./media_kit
  media_kit_video:
    git:
      url: https://github.com/media-kit/media-kit.git
      ref: main
      path: ./media_kit_video
  media_kit_libs_video:
    git:
      url: https://github.com/media-kit/media-kit.git
      ref: main
      path: ./libs/universal/media_kit_libs_video

dependency_overrides:
  media_kit:
    git:
      url: https://github.com/media-kit/media-kit.git
      ref: main
      path: ./media_kit
  media_kit_video:
    git:
      url: https://github.com/media-kit/media-kit.git
      ref: main
      path: ./media_kit_video
  media_kit_libs_video:
    git:
      url: https://github.com/media-kit/media-kit.git
      ref: main
      path: ./libs/universal/media_kit_libs_video
```

**Comment by DaVinci42** [comment]
@birros 

just reproduced on main branch with commit `3c899c`, I'm using iOS emulator

```
Thread 29 Crashed:
0   libsystem_kernel.dylib        	       0x1b17e40cc __pthread_kill + 8
1   libsystem_pthread.dylib       	       0x1b18381e8 pthread_kill + 256
2   libsystem_c.dylib             	       0x18012873c abort + 124
3   libc++abi.dylib               	       0x18023d0a8 abort_message + 128
4   libc++abi.dylib               	       0x18022dad8 demangling_terminate_handler() + 48
5   libc++abi.dylib               	       0x18023c580 std::__terminate(void (*)()) + 12
6   libc++abi.dylib               	       0x18023c51c std::terminate() + 32
7   media_kit_native_event_loop   	       0x10363436c std::__1::default_delete<std::__1::thread>::operator()[abi:v15006](std::__1::thread*) const + 52 (unique_ptr.h:48)
8   media_kit_native_event_loop   	       0x103634328 std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> >::reset[abi:v15006](std::__1::thread*) + 104 (unique_ptr.h:305)
9   media_kit_native_event_loop   	       0x1036342b0 std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> >::~unique_ptr[abi:v15006]() + 32 (unique_ptr.h:259)
10  media_kit_native_event_loop   	       0x10362a0c0 std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> >::~unique_ptr[abi:v15006]() + 28 (unique_ptr.h:259)
11  media_kit_native_event_loop   	       0x103635d8c std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >::~pair() + 32 (pair.h:40)
12  media_kit_native_event_loop   	       0x103635d5c std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >::~pair() + 28 (pair.h:40)
13  media_kit_native_event_loop   	       0x103635d00 void std::__1::allocator_traits<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > >::destroy[abi:v15006]<std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void, void>(std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> >&, std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >*) + 28 (allocator_traits.h:319)
14  media_kit_native_event_loop   	       0x103635ca0 std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > >::operator()[abi:v15006](std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>*) + 84 (__hash_table:815)
15  media_kit_native_event_loop   	       0x103635c3c std::__1::unique_ptr<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>, std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > > >::reset[abi:v15006](std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>*) + 104 (unique_ptr.h:305)
16  media_kit_native_event_loop   	       0x103635bc4 std::__1::unique_ptr<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>, std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > > >::~unique_ptr[abi:v15006]() + 32 (unique_ptr.h:259)
17  media_kit_native_event_loop   	       0x103634cec std::__1::unique_ptr<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>, std::__1::__hash_node_destructor<std::__1::allocator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*> > > >::~unique_ptr[abi:v15006]() + 28 (unique_ptr.h:259)
18  media_kit_native_event_loop   	       0x103639340 std::__1::__hash_table<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::__unordered_map_hasher<mpv_handle*, std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::hash<mpv_handle*>, std::__1::equal_to<mpv_handle*>, true>, std::__1::__unordered_map_equal<mpv_handle*, std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::equal_to<mpv_handle*>, std::__1::hash<mpv_handle*>, true>, std::__1::allocator<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > > > >::erase(std::__1::__hash_const_iterator<std::__1::__hash_node<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, void*>*>) + 96 (__hash_table:2403)
19  media_kit_native_event_loop   	       0x1036390ac unsigned long std::__1::__hash_table<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::__unordered_map_hasher<mpv_handle*, std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::hash<mpv_handle*>, std::__1::equal_to<mpv_handle*>, true>, std::__1::__unordered_map_equal<mpv_handle*, std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > >, std::__1::equal_to<mpv_handle*>, std::__1::hash<mpv_handle*>, true>, std::__1::allocator<std::__1::__hash_value_type<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > > > >::__erase_unique<mpv_handle*>(mpv_handle* const&) + 120 (__hash_table:2435)
20  media_kit_native_event_loop   	       0x1036385d4 std::__1::unordered_map<mpv_handle*, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> >, std::__1::hash<mpv_handle*>, std::__1::equal_to<mpv_handle*>, std::__1::allocator<std::__1::pair<mpv_handle* const, std::__1::unique_ptr<std::__1::thread, std::__1::default_delete<std::__1::thread> > > > >::erase[abi:v15006](mpv_handle* const&) + 32 (unordered_map:1360)
21  media_kit_native_event_loop   	       0x103638430 MediaKitEventLoopHandler::Dispose(long long)::$_1::operator()() const + 116 (media_kit_native_event_loop.cc:137)
22  media_kit_native_event_loop   	       0x103638388 decltype(std::declval<MediaKitEventLoopHandler::Dispose(long long)::$_1>()()) std::__1::__invoke[abi:v15006]<MediaKitEventLoopHandler::Dispose(long long)::$_1>(MediaKitEventLoopHandler::Dispose(long long)::$_1&&) + 24 (invoke.h:394)
23  media_kit_native_event_loop   	       0x103638364 void std::__1::__thread_execute[abi:v15006]<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, MediaKitEventLoopHandler::Dispose(long long)::$_1>(std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, MediaKitEventLoopHandler::Dispose(long long)::$_1>&, std::__1::__tuple_indices<>) + 28 (thread:290)
24  media_kit_native_event_loop   	       0x103638074 void* std::__1::__thread_proxy[abi:v15006]<std::__1::tuple<std::__1::unique_ptr<std::__1::__thread_struct, std::__1::default_delete<std::__1::__thread_struct> >, MediaKitEventLoopHandler::Dispose(long long)::$_1> >(void*) + 84 (thread:301)
25  libsystem_pthread.dylib       	       0x1b18384e4 _pthread_start + 116
26  libsystem_pthread.dylib       	       0x1b18336cc thread_start + 8
```

**Comment by DaVinci42** [comment]
please use this layout, I re-arranged the buttons to keep them in the same position so you don't need to move the mouse, which makes it easier to reproduce

```
import 'package:flutter/material.dart';
import 'package:media_kit/media_kit.dart';
import 'package:media_kit_video/media_kit_video.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
        useMaterial3: true,
      ),
      home: const Scaffold(
        body: SafeArea(child: PageA()),
      ),
    );
  }
}

class PageA extends StatefulWidget {
  const PageA({super.key});

  @override
  State<PageA> createState() => _PageAState();
}

class _PageAState extends State<PageA> {
  @override
  Widget build(BuildContext context) {
    return Center(
      child: Column(
        mainAxisSize: MainAxisSize.min,
        crossAxisAlignment: CrossAxisAlignment.center,
        children: [
          const SizedBox(
            width: 200,
            height: 200,
          ),
          ElevatedButton(
            onPressed: () {
              Navigator.of(context).push(
                MaterialPageRoute(builder: (_) => const MediaPage()),
              );
            },
            child: const Text("Open Media"),
          ),
        ],
      ),
    );
  }
}

class MediaPage extends StatefulWidget {
  const MediaPage({super.key});

  @override
  State<MediaPage> createState() => _MediaPageState();
}

class _MediaPageState extends State<MediaPage> {
  late final player = Player();
  late final controller = VideoController(player);

  @override
  void initState() {
    super.initState();
    MediaKit.ensureInitialized();
    player.open(
        Media(
            'https://user-images.githubusercontent.com/28951144/229373695-22f88f13-d18f-4288-9bf1-c3e078d83722.mp4'),
        play: false);
  }

  @override
  void dispose() {
    player.dispose();
    super.dispose();
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      body: SafeArea(
        child: Center(
          child: Column(
            mainAxisSize: MainAxisSize.min,
            crossAxisAlignment: CrossAxisAlignment.center,
            children: [
              SizedBox(
                width: 200,
                height: 200,
                child: Video(controller: controller),
              ),
              ElevatedButton(
                onPressed: () {
                  Navigator.of(context).pop();
                },
                child: const Text("POP"),
              ),
            ],
          ),
        ),
      ),
    );
  }
}

```

**Comment by birros** [comment]
I can reproduce on macOS.

**Comment by birros** [comment]
Try the following patch:

```diff
@@ -3,6 +3,7 @@ import 'package:media_kit/media_kit.dart';
 import 'package:media_kit_video/media_kit_video.dart';
 
 void main() {
+  MediaKit.ensureInitialized();
   runApp(const MyApp());
 }
 
@@ -71,7 +72,6 @@ class _MediaPageState extends State<MediaPage> {
   @override
   void initState() {
     super.initState();
-    MediaKit.ensureInitialized();
     player.open(
         Media(
             'https://user-images.githubusercontent.com/28951144/229373695-22f88f13-d18f-4288-9bf1-c3e078d83722.mp4'),
```

**Comment by alexmercerind** [comment]
Yeah... that method call actually frees previously allocated threads & mutexes, very expected.

**Comment by alexmercerind** [comment]
I'm going to close this as recommended solution is posted by @ birros.

**Comment by DaVinci42** [comment]
Thanks guys, confirm the issue is fixed.

But I have another concern, for a complex Flutter mobile app with 1 video page, 
we're not supposed to initialize the video library when the app starts, instead, should move to the place right before the usage.
in this way:
1. it won't impact the app startup time
2. it also avoids unnecessary memory usage, since there's the chance that the user never opens the video page.

basically, for all libraries, we should initialize in this lazy way.

but it's beyond the scope of this ticket.

**Comment by alexmercerind** [comment]
Then you only initialise it only for the very first time e.g. a Boolean value, not every single time page is mounted.

**Comment by alexmercerind** [comment]
For further details: The method not only does the "initialization" but also clears previously running `Player`s & `VideoController`s in order to handle hot-restart.
